rust/operator-binary/src/hdfs_controller.rs

```diff
@@ -233,8 +243,8 @@ pub async fn reconcile_hdfs(hdfs: Arc<HdfsCluster>, ctx: Arc<Ctx>) -> HdfsOperat
         .spec
         .image
         .resolve(DOCKER_IMAGE_BASE_NAME, crate::built_info::CARGO_PKG_VERSION);
-    if hdfs.has_kerberos_enabled() {
-        kerberos::check_if_supported(&resolved_product_image)?;
+    if hdfs.has_kerberos_enabled() && kerberos::check_if_supported(&resolved_product_image) {
```

Comment:
I would assume that `check_if_supported` returns `true` on success (though `is_supported` would arguably be clearer).

Reply:
Renamed in 8b25f33ab422c3fd908dc1de226e6896f185e4d3

Reply:
Fixed in e7160ad97d7beb9d96e057a07b3e0d87e993805e